server/store/user_store.go

```diff
@@ -278,3 +288,86 @@ func (index UserIndex) GetMattermostUserIDs() []string {
 
 	return result
 }
+
+// Get the check user status function
+func GetCheckUserStatus(store Store, logger bot.Logger, mattermostUserID string) func() bool {
```

Comment:
Maybe instead of `Get*`, we call these functions `Make*`, since we are creating and returning a new closure when these are called, rather than "getting" some pre-existing thing.